### ISSUE TITLE
No S3 permissions for LoginNodes CustomActions

### ISSUE THREAD

**Issue by michaelmayer2** [aws/aws-parallelcluster#6507]
I am very excited to see AWS ParallelCluster to have Support for CustomActions on the Login Nodes as of 3.11.0. 

There however is no way on how to add `Iam` `S3Access` permissions in the `LoginNodes` section of the `cluster-config.yaml`. 

It would be great to add this so that the CustomActions on the Login Nodes can be used more flexibly.

**Comment by hanwen-pcluste** [comment]
Hi Michael,

This task is in our backlog.

Thank you for the suggestion,
Hanwen

**Comment by joehellmersNOAA** [comment]
This is actually pretty important to us as well.

**Comment by michaelmayer2** [comment]
@hanwen-pcluste thanks so much for having this task in your backlog to make parallelcluster better. 

On a related note - not sure if it deserves a separate issue - at the moment when trying to use `CustomActions` with `LoginNodes` and there is any error in the script (for now it only cannot be downloaded, but later on I can imagine that the cript may contain errors and fail as well) the ScalingGroup is in a perpetual state of marking an instance failed and booting up a new one *without* a timeout. `pcluster describe-cluster` would still report `CREATION_IN_PROGRESS`. I had left one of my deployments running over night and when I eventually found out I could see the perpetual spin up and fail of the login nodes in the Activity Log of the ScalingGroup. It would be good to add timeouts to these operations as well.

**Comment by hanwen-pcluste** [comment]
My pleasure Michael!

Adding timeout or "[protected mode](https://docs.aws.amazon.com/parallelcluster/latest/ug/slurm-protected-mode-v3.html)" to login nodes is a great idea. Our team will discuss this.

Thank you,
Hanwen

**Comment by gmarciani** [comment]
Protected mode for login nodes has been added to our backlog

**Comment by gmarciani** [comment]
Hi @michaelmayer2  and @joehellmersNOAA ,

to customize IAM permissions for login nodes, you can specify additional IAM permissions for login nodes using `AdditionalIamPolicies`, which is even more powerful as it can add whatever additional permissions, not only S3.
See  https://docs.aws.amazon.com/parallelcluster/latest/ug/LoginNodes-v3.html#yaml-LoginNodes-Pools-Iam-AdditionalIamPolicies.

Would this be ok to address your use case?

**Comment by joehellmersNOAA** [comment]
I'll give it a shot.

**Comment by michaelmayer2** [comment]
`AdditionalIamPolicies` will definitely work, no doubt. I only raised the request to retain a bit of consistency in the YAML file. 

It is a bit unfortunate that S3 access handling between Head Node, Compute Nodes and Login Node Pools is not overly consistent. 

For the `HeadNode` section you have 

```
S3Access]:
      - BucketName: my-s3-buck
        EnableWriteAccess: true
        KeyName: my-key
```

which will take away the creation of IAM policies for S3 access while in order to use S3 access for Login node pools a user now has to create those policies in advance. 

But, again: `AdditionalIamPolicies` will technically work.

**Comment by joehellmersNOAA** [comment]
I agree with @michaelmayer2, it would be much better for consistency.
